qutebrowser/browser/commands.py

```diff
@@ -1214,23 +1214,43 @@ def bookmark_del(self, url=None):
 
     @cmdutils.register(instance='command-dispatcher', name='inspector',
                        scope='window')
-    def toggle_inspector(self):
+    @cmdutils.argument('position', choices=['right', 'left', 'top', 'bottom',
+                                            'window'])
+    def toggle_inspector(self, position=None):
         """Toggle the web inspector.
 
         Note: Due a bug in Qt, the inspector will show incorrect request
         headers in the network tab.
+
+        Args:
+            position: Where to open the inspector
+                      (right/left/top/bottom/window).
         """
         tab = self._current_widget()
         # FIXME:qtwebengine have a proper API for this
         page = tab._widget.page()  # pylint: disable=protected-access
 
+        @pyqtSlot()
+        def _on_inspector_closed():
+            tab.data.inspector = None
+
         try:
             if tab.data.inspector is None:
-                tab.data.inspector = inspector.create()
+                if position is None:
+                    try:
+                        position = configfiles.state['general'][
+                            'inspector_last_position']
+                    except KeyError:
+                        position = 'right'
```

Comment:
`position = configfiles.state['general'].get('inspector_last_position', 'right')` is shorter and may be clearer (assuming section `general` always exist)

Reply:
@The-Compiler does section `general` always exist?
proposal looks nice, thanks!

Reply:
Yup!

https://github.com/qutebrowser/qutebrowser/blob/5337d2cf31cdba0d2077cdfd86df8d116c3767ca/qutebrowser/config/configfiles.py#L68-L72